src/components/DateField/DateFieldInt.part.tsx

```diff
@@ -262,17 +262,16 @@ class DateFieldInt extends React.Component<DateFieldProps, DateFieldState> {
     const { onChange, name = '', form } = this.props;
 
     if (!this.valueControlled) {
+      this.setState({
+        value,
+        date,
+      });
       if (form) {
         form.change({
           name,
           value,
         });
       } else {
-        this.setState({
-          value,
-          date,
-        });
-
         this.onOpenChange(true);
       }
```

Comment:
could you consider this approach?

```tsx
if (!this.valueControlled) {
  this.setState({ value, date }, () => {
    form &&
      form.change({
        name,
        value,
      });
    this.onOpenChange(true);
  });
}
```

IMO: it looks much more readable...

Reply:
Done.